### ISSUE TITLE
OpenAPI Conformance error

### ISSUE THREAD

**Issue by justb4** [geopython/pygeoapi#421]
**Description**
Observed pygeoapi OpenAPI doc conformance error within GeoHealthCheck suddenly, after one of the last commits.

**Steps to Reproduce**
* First observed in GeoHealthCheck on https://demo.pygeoapi.io/master on both GHC demo server and GHC unit tests (which run OAS full validation Probe on https://demo.pygeoapi.io/master )
* GHC uses `openapi-spec-validator==0.2.8`

**Expected behavior**
OpenAPI doc should validate ok.

**Screenshots/Tracebacks**
Very hard to pinpoint, here is output from GHC:

```
2020-04-25 15:45:19,119 - GeoHealthCheck.probe - INFO - Result: success=False msg={'explode': False, 'in': 'query', 'name': 'id', 'required': False, 'schema': {'type': {'type': 'string'}}, 'style': 'form'} is not valid under any of the given schemas

Failed validating 'oneOf' in schema['properties']['paths']['patternProperties']['^/']['properties']['get']['properties']['parameters']['items']:
    {'oneOf': [{'$ref': '#/definitions/parameter'},
               {'$ref': '#/definitions/reference'}]}

On instance['paths']['/collections/lakes/items']['get']['parameters'][5]:
    {'explode': False,
     'in': 'query',
     'name': 'id',
     'required': False,
     'schema': {'type': {'type': 'string'}},
     'style': 'form'} response_time=7.376262
resource: https://demo.pygeoapi.io/master result=False
```

and 
```
Hi: this is an automated message from the GeoHealthCheck Demonstration service.

Resource: OGC WFS3
Resource type: OGC:WFS3
Resource URL: https://demo.pygeoapi.io/master

Details:

Date: 2020-04-25T13:34:46Z 

Message: {'explode': False, 'in': 'query', 'name': 'id', 'required': False, 'schema': {'type': {'type': 'string'}}, 'style': 'form'} is not valid under any of the given schemas

Failed validating 'oneOf' in schema['properties']['paths']['patternProperties']['^/']['properties']['get']['properties']['parameters']['items']:
    {'oneOf': [{'$ref': '#/definitions/parameter'},
               {'$ref': '#/definitions/reference'}]}

On instance['paths']['/collections/lakes/items']['get']['parameters'][5]:
    {'explode': False,
     'in': 'query',
     'name': 'id',
     'required': False,
     'schema': {'type': {'type': 'string'}},
     'style': 'form'}
```

**Environment**
 - OS: any
 - Python version: 3
 - pygeoapi version: `master`

**Additional context**

* no change to GeoHealthCheck nor its OWSLib version was made in months
* No errors when using the `pygeoapi` stable version https://demo.pygeoapi.io/stable
* could also be related to the remote OGC OA schemas...
* could also be same error as current Travis build failure on `pygeoapi` `master..`

Very hard to pinpoint, more investigation needed.

**Comment by justb4** [comment]
Maybe it is the `stac` part? Here some output from GHC:

```
2020-04-25 20:10:45,685 - openapi_spec_validator.validators - DEBUG - Dereferencing {'get': {'description': 'SpatioTemporal Asset Catalog', 'parameters': [], 'responses': {'200': {'$ref': '#/components/responses/200', 'x-scope': ['']}, 'default': {'$ref': '#/components/responses/default', 'x-scope': ['']}}, 'summary': 'SpatioTemporal Asset Catalog', 'tags': ['stac']}}
2020-04-25 20:10:45,685 - openapi_spec_validator.validators - DEBUG - Dereferencing {'get': {'description': 'SpatioTemporal Asset Catalog', 'parameters': [], 'responses': {'200': {'$ref': '#/components/responses/200', 'x-scope': ['']}, 'default': {'$ref': '#/components/responses/default', 'x-scope': ['']}}, 'summary': 'SpatioTemporal Asset Catalog', 'tags': ['stac']}}
2020-04-25 20:10:45,686 - openapi_spec_validator.validators - DEBUG - Dereferencing {'description': 'SpatioTemporal Asset Catalog', 'parameters': [], 'responses': {'200': {'$ref': '#/components/responses/200', 'x-scope': ['']}, 'default': {'$ref': '#/components/responses/default', 'x-scope': ['']}}, 'summary': 'SpatioTemporal Asset Catalog', 'tags': ['stac']}
2020-04-25 20:10:45,686 - openapi_spec_validator.validators - DEBUG - Dereferencing {'parameters': {'f': {'description': 'The optional f parameter indicates the output format which the server shall provide as part of the response document.  The default format is GeoJSON.', 'explode': False, 'in': 'query', 'name': 'f', 'required': False, 'schema': {'default': 'json', 'enum': ['json', 'html', 'jsonld'], 'type': 'string'}, 'style': 'form'}, 'sortby': {'description': 'The optional sortby parameter indicates the sort property and order on which the server shall present results in the response document using the convention `sortby=PROPERTY:X`, where `PROPERTY` is the sort property and `X` is the sort order (`A` is ascending, `D` is descending). Sorting by multiple properties is supported by providing a comma-separated list.', 'explode': False, 'in': 'query', 'name': 'sortby', 'required': False, 'schema': {'type': 'string'}, 'style': 'form'}, 'startindex': {'description': 'The optional startindex parameter indicates the index within the result set from which the server shall begin presenting results in the response document.  The first element has an index of 0 (default).', 'explode': False, 'in': 'query', 'name': 'startindex', 'required': False, 'schema': {'default': 0, 'minimum': 0, 'type': 'integer'}, 'style': 'form'}}, 'responses': {'200': {'description': 'successful operation'}, 'default': {'content': {'application/json': {'schema': {'$ref': 'https://raw.githubusercontent.com/opengeospatial/wps-rest-binding/master/core/openapi/schemas/exception.yaml', 'x-scope': ['', '#/components/responses/default']}}}, 'description': 'Unexpected error'}}}
2020-04-25 20:10:45,686 - openapi_spec_validator.validators - DEBUG - Dereferencing {}
2020-04-25 20:10:45,687 - GeoHealthCheck.probe - INFO - Result: success=False msg={'explode': False, 'in': 'query', 'name': 'id', 'required': False, 'schema': {'type': {'type': 'string'}}, 'style': 'form'} is not valid under any of the given schemas

Failed validating 'oneOf' in schema['properties']['paths']['patternProperties']['^/']['properties']['get']['properties']['parameters']['items']:
    {'oneOf': [{'$ref': '#/definitions/parameter'},
               {'$ref': '#/definitions/reference'}]}

On instance['paths']['/collections/lakes/items']['get']['parameters'][5]:
    {'explode': False,
     'in': 'query',
     'name': 'id',
     'required': False,
     'schema': {'type': {'type': 'string'}},
     'style': 'form'} response_time=9.932113

```

**Comment by tomkralidis** [comment]
Fixed in via 11ee7053fc374b106225d84a3d547419eaf0e938 and 6916099315f77a53b12e3733b883e609f33a81e1

**Comment by justb4** [comment]
Great! Also got the "Fixed" email from GHC Demo. Thanks @tomkralidis !
